core/tee/tee_svc_cryp.c

```diff
@@ -517,18 +517,26 @@ static void op_u32_to_binary_helper(uint32_t v, uint8_t *data,
 				    size_t data_len, size_t *offs)
 {
 	uint32_t field;
+	size_t temp;
+
+	if (ADD_OVERFLOW(*offs, sizeof(field), &temp))
+		return;
 
 	if (data && (*offs + sizeof(field)) <= data_len) {
```

Comment:
Since we've already calculated `*offs + sizeof(field)` use that value instead.

Reply:
OK